chain/sui/tx.go

```diff
@@ -239,3 +240,85 @@ func (tx Tx) GetMetadata() ([]byte, error) {
 	}
 	return metadataBz, nil
 }
+
+type StakingInput struct {
+	TxInput
+	Validator string `json:"validator"`
+}
+
+var _ xc.TxVariantInput = &StakingInput{}
+var _ xc.StakeTxInput = &StakingInput{}
+
+func (*StakingInput) Staking() {}
+func (*StakingInput) GetVariant() xc.TxVariantInputType {
+	return xc.NewStakingInputType(xc.DriverSui, string(xc.Native))
+}
+
+type Stake struct {
+	Principal xc.AmountBlockchain `json:"principal"`
+	Rewards   xc.AmountBlockchain `json:"rewards"`
+	ObjectId  string              `json:"object_id"`
+	Version   uint64              `json:"version"`
+	Digest    string              `json:"digest"`
+	State     xclient.StakeState  `json:"state"`
+	Validator string              `json:"validator"`
```

Comment:
This is `repeated`, from `StakingInput.Validator`, is it simplified?  Also don't want to replace user input `StakingArgs.Validator` on the transaction builder.